torch/onnx/_internal/fx/torch_export_graph_extractor.py

```diff
@@ -0,0 +1,72 @@
+# NOTE: This file is referenced by name at
+#       /opt/pytorch/torch/_dynamo/eval_frame.py::DONT_WRAP_FILES.
+#       introduced by https://github.com/pytorch/pytorch/pull/98894.
+#       If this file is renamed, moved, etc please update the reference there!
+
+from __future__ import annotations
+
+from typing import Any, Callable, Mapping, Optional, Sequence, Union
+
+import torch._dynamo
+import torch.fx
+import torch.onnx
+from torch.onnx._internal import _beartype, exporter, io_adapter
+
+
+class TorchExport(exporter.FXGraphExtractor):
+    """Generates a FX GraphModule using torch.export API
+    Args:
+        aten_graph: If True, exports a graph with ATen operators.
+                    If False, exports a graph with Python operators.
+    """
+
+    def __init__(
+        self,
+        aten_graph: Optional[bool] = None,
+    ):
+        super().__init__()
+        self.aten_graph = aten_graph or True
+
+    def generate_fx(
+        self,
+        options: exporter.ResolvedExportOptions,
+        model: "ExportedProgram",  # type: ignore[name-defined]
+        model_args: Sequence[Any],
+        model_kwargs: Mapping[str, Any],
+    ) -> torch.fx.GraphModule:
+        # No need to translate callable to FX graph.
+        # This FX Graph extractor assumes `model` was obtained through
+        #     exported_program = torch.export.export(
+        #         model,
+        #         args=model_args,  # type: ignore[arg-type]
+        #         kwargs=model_kwargs,  # type: ignore[arg-type]
+        #     )
+
+        # Export FX graph to ONNX ModelProto.
+        self.input_adapter.append_step(
+            io_adapter.FlattenInputWithTreeSpecValidationInputStep()
+        )
+        self.input_adapter.append_step(
+            io_adapter.PrependParamsAndBuffersAotAutogradInputStep()
+        )
+
+        updated_model_args = self.input_adapter.apply(
+            *model_args, model=model, **model_kwargs
+        )
+
+        # Export FX graph to ONNX ModelProto.
+        return self.pre_export_passes(options, model, model.graph_module, updated_model_args)  # type: ignore[return-value]
+
+    @_beartype.beartype
+    def pre_export_passes(
```

Comment:
Some passes still needed I believe, what is the plan for follow-ups?

Reply:
Any in particular comes to mind?
I am adding them as I discover the need for them.  Right now there is only one simple test, so maybe some passes are missed

My idea was to create a PR in which we can reuse all `TestFxToOnnxWithOnnxRuntime` for testing with both `fx_tracer=dynamo_export_graph_extractor.DynamoExport` and `fx_tracer=torch_export_graph_extractor.TorchExport`. That would flush everything we need to support

Reply:
type promotion is immediate one I can think of. Yep we can play catch-up with test coverage and bench results.